packages/core/mesh/teleport/src/muxing/framer.ts

```diff
@@ -50,11 +62,19 @@ export class Framer {
     send: (message) => {
       // log('write', { len: message.length, frame: Buffer.from(message).toString('hex') })
       return new Promise<void>((resolve) => {
-        const canContinue = this._stream.push(encodeFrame(message));
-        if (!canContinue) {
-          this._responseQueue.push(resolve);
-        } else {
-          process.nextTick(resolve);
+        const isEmpty = this._chunkQueue.length === 0;
+        const data = encodeFrame(message);
+
+        const chunks = [];
+        for (let i = 0; i < data.length; i += MAX_CHUNK_SIZE) {
+          chunks.push(data.slice(i, i + MAX_CHUNK_SIZE));
+        }
+        chunks.forEach((chunk, index) => {
+          const callback = index === chunks.length - 1 ? resolve : undefined;
+          this._chunkQueue.push({ msg: chunk, callback });
+        });
+        if (isEmpty) {
+          this._processChunkQueue().catch(log.catch);
```

Comment:
call the logger in a function call `log.catch(....)` this way the build metadata (filename, line number) would be automatically added during compilation